lib/ansible/modules/cloud/azure/azure_rm_mysqlserver.py

```diff
@@ -0,0 +1,368 @@
+    ssl_enforcement:
+        description:
+            - "Enable ssl enforcement or not when connect to server. Possible values include: 'Enabled', 'Disabled'"
+    create_mode:
+        description:
+            - "Currently only 'Default' value supported"
+    admin_username:
+        description:
+            - "The administrator's login name of a server. Can only be specified when the server is being created (and is required for creation)."
+    admin_password:
+        description:
+            - The password of the administrator login.
+
+extends_documentation_fragment:
+    - azure
+    - azure_tags
+
+author:
+    - "Zim Kalinowski (@zikalino)"
+
+'''
+
+EXAMPLES = '''
+  - name: Create (or update) MySQL Server
+    azure_rm_mysqlserver:
+      resource_group: resource_group_name
+      name: server_name
+      sku:
+        name: name
+        tier: tier
+        capacity: capacity
+        size: size
+        family: family
+      location: location
+      storage_mb: storage_mb
+      version: version
+      ssl_enforcement: ssl_enforcement
+      create_mode: create_mode
+      admin_username: administrator_login
+      admin_password: administrator_login_password
+'''
+
+RETURN = '''
+id:
+    description:
+        - Resource ID
+    returned: always
+    type: str
+    sample: id
+version:
+    description:
+        - "Server version. Possible values include: '5.6', '5.7'"
+    returned: always
+    type: str
+    sample: version
+user_visible_state:
+    description:
+        - "A state of a server that is visible to user. Possible values include: 'Ready', 'Dropping', 'Disabled'"
+    returned: always
+    type: str
+    sample: user_visible_state
+fully_qualified_domain_name:
+    description:
+        - The fully qualified domain name of a server.
+    returned: always
+    type: str
+    sample: fully_qualified_domain_name
+'''
+
+import time
+from ansible.module_utils.azure_rm_common import AzureRMModuleBase
+
+try:
+    from msrestazure.azure_exceptions import CloudError
+    from msrestazure.azure_operation import AzureOperationPoller
+    from azure.mgmt.rdbms.mysql import MySQLManagementClient
+    from msrest.serialization import Model
+except ImportError:
+    # This is handled in azure_rm_common
+    pass
+
+
+class Actions:
+    NoAction, Create, Update, Delete = range(4)
+
+
+class AzureRMServers(AzureRMModuleBase):
+    """Configuration class for an Azure RM MySQL Server resource"""
+
+    def __init__(self):
+        self.module_arg_spec = dict(
+            resource_group=dict(
+                type='str',
+                required=True
+            ),
+            name=dict(
+                type='str',
+                required=True
+            ),
+            sku=dict(
+                type='dict',
+                required=False
+            ),
+            location=dict(
+                type='str',
+                required=False
+            ),
+            storage_mb=dict(
+                type='long',
```

Comment:
`long` isn't a valid type in Python see here https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/basic.py#L854-L867 for supported types in ansible.